control-plane/csi-driver/src/bin/node/node.rs

```diff
@@ -367,23 +368,29 @@ impl node_server::Node for Node {
             ));
         }
         match sys::statfs::statfs(&*msg.volume_path) {
-            Ok(info) => Ok(Response::new(NodeGetVolumeStatsResponse {
-                usage: vec![
-                    csi::VolumeUsage {
-                        total: info.blocks() as i64 * info.block_size(),
-                        unit: csi::volume_usage::Unit::Bytes as i32,
-                        available: info.blocks_available() as i64 * info.block_size(),
-                        used: (info.blocks() - info.blocks_free()) as i64 * info.block_size(),
-                    },
-                    csi::VolumeUsage {
-                        total: info.files() as i64,
-                        unit: csi::volume_usage::Unit::Inodes as i32,
-                        available: info.files_free() as i64,
-                        used: (info.files() - info.files_free()) as i64,
-                    },
-                ],
-                volume_condition: None,
-            })),
+            Ok(info) => match info.filesystem_type() {
+                sys::statfs::EXT4_SUPER_MAGIC | FsType(XFS_MAGIC) => {
+                    Ok(Response::new(NodeGetVolumeStatsResponse {
+                        usage: vec![
```

Comment:
@tiagolobocastro @Abhinandan-Purkait I really had trouble writing it by hand and I'm sure reading this is confusing as well, so I'm suggesting two ways to probably write it in a better-rust-way.

1. Use the `map` way as suggested by [this StackOverflow answer](https://stackoverflow.com/a/38523776) or just drop-braces with which I'm not super comfortable atm?

2. Use `ok_or` for `Result<T, err>` types as suggested [here](https://users.rust-lang.org/t/nested-match-hell-in-rust/57628/10) and use the match for matching against `filesystem_type()`.